src/main/java/Bot.java

```diff
@@ -0,0 +1,296 @@
+import java.util.ArrayList;
+import java.util.Scanner;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.List;
+
+
+public class Bot {
+    private final String filePath = "./data/Venti.txt";
+    ArrayList<Task> taskList;
+    public Bot() {
+
+        taskList=new ArrayList<Task>();
+        createFileIfNotExist();
+        loadTasksFromFile();
+    }
+    private void echo(String st)
+    {
+        System.out.println("added: "+st);
+        printLine();
+    }
+    private void createFileIfNotExist() {
+        try {
+            File file = new File(filePath);
+            file.getParentFile().mkdirs(); // Create directory if it doesn't exist
+            file.createNewFile(); // Create file if it doesn't exist
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private void loadTasksFromFile() {
+        try {
+            List<String> lines = Files.readAllLines(Paths.get(filePath));
+            for (String line : lines) {
+                // Assuming the format is: Type | Status | Description | [Additional Info]
+                String[] parts = line.split(" \\| ");
+                switch (parts[0]) {
+                    case "T":
+                        taskList.add(new Todo(parts[2],parts[1].equals("1")));
+                        break;
+                    case "D":
+                        taskList.add(new Deadline(parts[2], parts[3], parts[1].equals("1")));
+                        break;
+                    case "E":
+                        taskList.add(new Event(parts[2], parts[3],parts[4], parts[1].equals("1")));
+                        break;
+                }
+            }
+        } catch (IOException e) {
+            System.out.println("Error loading tasks from file.");
+        }
+    }
+
+    private void saveTasksToFile() {
+        try (FileWriter writer = new FileWriter(filePath)) {
+            for (Task task : taskList) {
+                writer.write(task.toFileFormat() + "\n");
+            }
+        } catch (IOException e) {
+            System.out.println("Error saving tasks from file.");
+        }
+    }
+
+    private void printLine()
+    {
+        System.out.println("____________________________________________________________");
+    }
+    private void addList(Task t)
+    {
+        this.taskList.add(t);
+        saveTasksToFile();
+    }
+    private void sayBye()
+    {
+        System.out.println("Bye. Hope to see you again soon!");
+        printLine();
+    }
+    private void printList()
+    {
+        System.out.println("Here are the tasks in your list:");
+        for(int i=0;i<this.taskList.size();i++)
+        {
+            System.out.println((i+1)+". "+this.taskList.get(i).toString());
+        }
+        printLine();
+    }
+    private void markAsDone(int index)
+    {
+        if (index >= 0 && index < this.taskList.size()) {
```

Comment:
The code looks deeply nested. Perhaps you could refactor with a [guard clause](https://refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html)?

Same with your `markAsUndone` function